Address PR feedback

aditya8242 opened the pull request.

aditya8242 committed:
docs: standardize ini.h documentation to JavaDoc format

aditya8242 committed:
Address PR feedback

ccgargantua commented:
What exactly did you fix? Still does not build, big table of function forward decs still gone, still contains functions without completed javadoc comments. Dude, your diff should not be removing almost 500 loc. What isn't clicking here? Is something not clear?

ccgargantua commented:
Are you using AI?

aditya8242 commented:
Yeah, I did use AI as I thought these was as simple as adding comments.
Since this was marked as a `good first issue` I got excited and tried to use ChatGPT to do the job.

My apologies, I'll try to understand your whole project first, learn about JavaDocs, and maybe then I'll try to fix the issue.

ccgargantua commented:
Please don't bother. You obviously can't do it so save us both some time and just give up. Word of advice: If you are trying to contribute to open source projects, don't waste actual maintainers' time by opening two PRs in a row using AI both times, even when your first is a total failure that breaks the build. I told you to run the tests before you open another PR and linked you to the guidelines, and you can't take 60 seconds to run the tests for this simple project and see that it doesn't compile. Are you that lazy? It is a good first issue for beginners but you aren't even a beginner, you're a lazy grifter just wanting more numbers for your fancy README. Your README doesn't mean shit because you don't have a clue about contributing to real world projects, and you've proven that by failing at a task that would've taken you 10 minutes to do if you just did it yourself. Seriously, opening a second PR that still doesn't fucking build and does not address *any* of the issues with the first, then having the audacity to title it "Address PR feedback" is unbelievable. You are a joke of a programmer and will never achieve anything if you don't change your ways. You waste my time with your stupid AI garbage. Learn to code you moron.

aditya8242 commented:
@ccgargantua I know I deserve the criticism and I'll to handle it constructively. My apologies again.

I'm now reading your code and trying to understand it and actually test it first initially without any changes.

I'll try my best to read everything, test it out first, and then make a final try before giving up.

Hope you understand, its an itch I've created to myself, but I won't bother you before a last try.

Please calm down and maybe I'll learn a thing or two because of you.

aditya8242 commented:
<img width="1366" height="741" alt="image" src="https://github.com/user-attachments/assets/00a5ea87-b5c0-4f70-bf48-4cc76cbe9b18" />

Actually doing the stuff, I was lazy I admit it, you were right, it was the greed of fancy stuff on README.
Your project is really good for a beginner like me.

Please give me a chance, I won't disappoint.

ccgargantua commented:
You can open one more PR. If I feel it is AI generated, you will be blocked and restricted from contributing. To build the tests, provide CMake with a `-DINI_TEST=ON`. Run the resulting executable to run the tests.

aditya8242 commented:
Hello again @ccgargantua,

I rewrote the entire comments of `ini.h` in JavaDoc style by myself, no AI, just a bit of Googling.

I fiddled with the code and tried to understand what the project is for and what ini files are.
Not just for the sake of doing open source contributions and just for adding comments.

I learned a lot of new things in C such as how generic programming is done in C, `#define` functions, etc.

So before moving forward and creating another PR I tested the code multiple times and all tests are passing.

# Command used for testing:
1. `cmake . -DINI_TEST=1` as mentioned by you.
2. `cmake --build build`
3. `./build/ini_tests`

<img width="1366" height="741" alt="image" src="https://github.com/user-attachments/assets/414d9e03-b3ff-42be-9c28-2ff210d7870f" />

Please review this and give me a green signal.

Also added three lines between sections.

You can tell me to modify it later on.

ccgargantua commented:
If all you say is true then go ahead and open the PR.
